### ISSUE TITLE
Get rid of chcp

### ISSUE THREAD

**Issue by sagotch** [ocsigen/ocsigen-start#419]
Starting a new project with ocsigen-start (actually trying to upgrade old code using EBA), I find it to difficult to get rid of everything I do not want, in particular, I have no idea about how to remove this awful chcp thing...

**Comment by dannywillems** [comment]
I don't understand why you want to remove it.
If you don't want to use CHCP to update your app without update a new app on the store, you only regenerate the app archive and upload it.

**Comment by sagotch** [comment]
Obviously, but why should I embed/execute the code I do not need in my application?

**Comment by dannywillems** [comment]
Oh, I forgot there were some redirection to `eliom.html`.

As CHCP is only executed in `eliom_loader.ml`, I think removing `index.html` and `eliom_loader.ml`, rename `eliom.html` in `index.html` and replace redirection `eliom.html` by `index.html` is enough.

I think we could imagine a solution like `make android-no-chcp` which does the job. I think it's not a lot of work.

**Comment by vasilisp** [comment]
Unfortunately, we do need CHCP (or equivalent). It is an artifact of how Eliom works. If you recompile the application, the service identifiers may change, so the client portion needs to be updated.

Waiting for the user to update via the app store for the app to function again is unacceptable; all but the most dedicated users would give up the first time this happens.

If there is a programmatic way to trigger an update via the app store, we could consider that.

**Comment by vasilisp** [comment]
What @dannywillems suggests could work. It would be convenient for development. But I wouldn't recommend it for the distributable version of the app :).

**Comment by sagotch** [comment]
What do you mean by "service identifiers may change"? How is this identifier defined?

**Comment by vasilisp** [comment]
When you call a pathless service (what used to be called a non-attached coservice), we use a special parameter containing an "identifier" to recognize the service and call the appropriate handler. This parameter is in the URL (for GET services) and in the request body for POST services, but in either case it is automatically generated (in an unpredictable way) when you launch the app.

Similarly, the implementation of client values and injections involves a lot of automatically-generated and unpredictable identifiers.

All these identifiers show up in the client-server interface and necessitate updates.

Now, you could follow a programming regime that doesn't use the above functionality, but only a small fragment of Eliom would remain. OS uses much more :).

**Comment by sagotch** [comment]
Does the `name` parameter of `Eliom_service.create` would be enough for the client side still call the right service even if the server is recompiled? Even for the parameters name?

**Comment by vasilisp** [comment]
Yes, that would work.

The problem with client values and injections remains. We could provide some mechanism for naming them, but there is nothing right now.

**Comment by sagotch** [comment]
Good to know. I think I will end up with code avoiding injections if possible, if the `name` param is enough for compatibility between old client / new server.

Also, I guess I will go for a lightweight homemade version of ocsigen-start... You can close the issue as it seems that it won't be fixed in ocsigen-start anyway.

Thanks

**Comment by vasilisp** [comment]
Note that there are injections in the code of Eliom itself (and in OS, OT, ...), and these may get in your way...

A lightweight OS (template?) is more than welcome. Keep us updated :).

I would prefer the issue to stay open. It is a valid issue, albeit not one that we can tackle immediately.
